호텔 조회 기능 수정

nasaoreo opened the pull request:
## 작업 내용
- 호텔 조회 기능을 skyscanner와 동일하게 호텔 단위로 변경 

## 수정 전/후 비교
### 수정 전
- 호텔 검색 결과는 **호텔 방** 목록
- 상세 조회는 **호텔 방**에 대해서 수행
- 상세 조회 결과는 **호텔 방 정보**

### 수정 후
- 호텔 검색 결과는 **호텔** 목록
- 상세 조회는 **호텔**에 대해서 수행
- 상세 조회 결과는 **호텔 정보 및 예약 가능한 방 목록**

Closes #22

nasaoreo committed:
Fix: 호텔 조회 기능 수정

- 호텔 검색 결과가 호텔 단위가 되도록 수정
- 상세 조회를 요청했을 때 호텔과 호텔의 예약 가능한 방 목록을 제공하도록 수정

dia312 commented on src/main/java/com/grepp/carrierroute/hotel/service/HotelService.java:
```diff
@@ -68,10 +66,26 @@ public HotelSearchResponseDto findRoomBy(Long id) throws HotelInfoNotFoundedExce
+        return hotel.getHotelRooms()
+                .stream()
+                .filter(room -> room.getMaxGuestNumber() >= requestDto.getGuestNumber())
+                .filter(room -> room.getCount() >= requestDto.getNumOfRoom())
```
해당 코드 보면서 숫자의 대소 비교를 함수로 비교하면 어떨까 .. 해서 compareTo를 사용하면 괜찮을까? 생각했는데 오히려 가독성이 떨어지겠구나,,생각해보았습니당 ㅋㅋ

nasaoreo committed:
Refactor: 호텔 조회 기능 리팩토링

- Wrapper 타입의 사용이 불가피한 경우 아니라면 primitive 타입으로 변경
- Entity, DTO 객체 생성 시 검증을 어노테이션 기반에서 메소드 기반으로 변경

nasaoreo commented:
코드 리뷰 완료 후, 팀원 전체 동의 하에 Merge 합니다!
